### DISCUSSION TITLE
How much time for tritonroute to run the benchmarks of ispd2018 and ispd2019?

### DISCUSSION THREAD

**Discussion by gilgamsh** [The-OpenROAD-Project/OpenROAD#3121] [Q&A]
I am using https://github.com/The-OpenROAD-Project/TritonRoute to run the benchmarks of ispd2018&2019. However I found Multithreaded acceleration is almost useless.  Does the legacy code in https://github.com/The-OpenROAD-Project/TritonRoute support Multithreaded acceleration ?

**Comment by maliberty** [comment]
How are you enabling the multi threading? It should make a large difference

**Reply by gilgamsh** [reply]
I use `-threads 8`.  
The whole command is like `./TritonRoute -lef <LEF_FILE> -def <DEF_FILE> -guide <GUIDE_FILE>  -threads 8  -output <OUTPUT_DEF> `
I think I have found the reason. I use gprof (add `-pg` when compiling) to profile the code. after remove the `-pg` I found it is superfast.

**Comment by maliberty** [comment]
The standalone TritonRoute is archived and not under any active development or support.  If that meets you needs that's fine but the integrated version in OR is continuing to be improved.

**Reply by gilgamsh** [reply]
Thank you very much

**Reply by gilgamsh** [reply]
Could you give me some basic documents about how to use Tritonroute in OR, since the document in https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/drt/README.md is almost empty? Thank you very much.

**Comment by maliberty** [comment]
Since you are interested in contests I would look at 
https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/drt/test/ispd18_sample.tcl
https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/drt/test/run-ispd.py

as examples

**Reply by gilgamsh** [reply]
Thank you very much.
